src/Common/ThreadPool.cpp

```diff
@@ -672,7 +672,7 @@ ThreadPoolImpl<Thread>::ThreadFromThreadPool::~ThreadFromThreadPool()
 template <typename Thread>
 void ThreadPoolImpl<Thread>::ThreadFromThreadPool::worker()
 {
-    DENY_ALLOCATIONS_IN_SCOPE;
+    // DENY_ALLOCATIONS_IN_SCOPE;
```

Comment:
Unfortunately, allocations are still happening. The first suspect is `DB::Exception::initializeThreadFramePointers();`, but need to dig deeper.